include/flexible.hxx

```diff
@@ -0,0 +1,464 @@
+    }
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator*=(BoutReal rhs) {
+    ((F*)fields[mainid])->operator*=(rhs);
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator/=(const Field3D & rhs) {
+    if (mainid == getId(rhs.getLocation())){
+      ((F*)fields[mainid])->operator/=(rhs);
+    } else {
+      throw BoutException("Not yet implemtented!");
+    }
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator/=(const Field2D & rhs) {
+    if (mainid == getId(rhs.getLocation())){
+      ((F*)fields[mainid])->operator/=(rhs);
+    } else {
+      throw BoutException("Not yet implemtented!");
+    }
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator/=(BoutReal rhs) {
+    ((F*)fields[mainid])->operator/=(rhs);
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator+=(const Field3D & rhs) {
+    if (mainid == getId(rhs.getLocation())){
+      ((F*)fields[mainid])->operator+=(rhs);
+    } else {
+      throw BoutException("Not yet implemtented!");
+    }
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator+=(const Field2D & rhs) {
+    if (mainid == getId(rhs.getLocation())){
+      ((F*)fields[mainid])->operator+=(rhs);
+    } else {
+      throw BoutException("Not yet implemtented!");
+    }
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator+=(BoutReal rhs) {
+    ((F*)fields[mainid])->operator+=(rhs);
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator-=(const Field3D & rhs) {
+    if (mainid == getId(rhs.getLocation())){
+      ((F*)fields[mainid])->operator-=(rhs);
+    } else {
+      throw BoutException("Not yet implemtented!");
+    }
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator-=(const Field2D & rhs) {
+    if (mainid == getId(rhs.getLocation())){
+      ((F*)fields[mainid])->operator-=(rhs);
+    } else {
+      throw BoutException("Not yet implemtented!");
+    }
+    clean(false);
+    return *this;
+  };
+  Flexible<F>& operator-=(BoutReal rhs) {
+    ((F*)fields[mainid])->operator-=(rhs);
+    clean(false);
+    return *this;
+  };
+
+private:
+  // Helper function to get index of location.
+  uint getId(CELL_LOC loc_){
+    uint loc = static_cast<uint>(loc_)-1;
+    if ( loc > num_fields || loc < 0){
+      throw BoutException("Unexpeted Fieldlocation! (Info: I got %d - %s)",loc,strLocation(loc_));
+    }
+    return loc;
+  };
+  void init(const F * main){
+    mainloc=main->getLocation();
+    mainid = getId(mainloc);
+    for (uint i=0;i<num_fields;++i){
+      fields[i]=nullptr;
+    }
+    fields[mainid]=main;
+    owner[mainid]=true;
+  };
+  void clean(bool include_main){
+    for (uint i=0;i<num_fields;i++){
+      if (i != mainid || include_main) {
+        if (fields[i] != nullptr){
+	  if (owner[i]){
+	    delete fields[i];
+	  }
+	  fields[i]=nullptr;
+        }
+      }
+    }
+  };
+  const F * geti(CELL_LOC loc_) {
```

Comment:
This is a very confusing name, what does it mean?

Reply:
Function has been removed, work is now done in `getNonConst(CELL_LOC)`